client/js/util.js

```diff
@@ -41,3 +41,31 @@ export {
   setPropertyPrefixed,
   featureTest
 };
+
+// Event delegation
+function checkTarget(delegateEl, eventEl, possibleTarget) {
+  if (eventEl === delegateEl) {
+    return;
+  } else if (eventEl === possibleTarget) {
+    return possibleTarget;
+  } else {
+    return checkTarget(delegateEl, eventEl.parentNode, possibleTarget);
+  }
+}
+
+export function on(delegateElSelector, eventName, eventElSelector, fn) {
+  const delegateEl = document.querySelector(delegateElSelector);
+
+  delegateEl.addEventListener(eventName, (event) => {
+    const possibleTargets = nodeList(delegateEl.querySelectorAll(eventElSelector));
+    const eventEl = event.target;
+
+    possibleTargets.forEach((possibleTarget) => {
+      const correctTarget = checkTarget(delegateEl, eventEl, possibleTarget);
+
+      if (correctTarget) {
+        return fn.call(correctTarget, event);
+      }
+    });
+  });
+}
```

Comment:
As this does a DOM read every time - do you think this might be a good place to add fastDOM?